$OpenBSD: patch-configure_d_config_os_headers,v 1.2 2011/01/15 09:38:31 sthen Exp $

On OpenBSD headers must be included in this order:

kernel
network
others

Also in_pcb.h requires ip.h

--- configure.d/config_os_headers.orig	Fri Nov  5 13:55:44 2010
+++ configure.d/config_os_headers	Wed Jan  5 17:13:57 2011
@@ -391,17 +391,20 @@ AC_CHECK_HEADERS(netinet/in_pcb.h,,,
 #if HAVE_SYS_SOCKETVAR_H
 #include <sys/socketvar.h>
 #endif
-#if HAVE_ARPA_INET_H
-#include <arpa/inet.h>
+#if HAVE_NETINET_IN_H
+#include <netinet/in.h>
 #endif
 #if HAVE_NETINET_IN_SYSTM_H
 #include <netinet/in_systm.h>
 #endif
-#if HAVE_NETINET_IN_H
-#include <netinet/in.h>
+#if HAVE_NETINET_IP_H
+#include <netinet/ip.h>
 #endif
 #if HAVE_NET_ROUTE_H
 #include <net/route.h>
+#endif
+#if HAVE_ARPA_INET_H
+#include <arpa/inet.h>
 #endif
     ]])
 
